### ISSUE TITLE
Compatibility with Pinocchio on MacOS 13

### ISSUE THREAD

**Issue by zongmianli** [stack-of-tasks/eigenpy#19]
I bound a C++ function `foo` with `Eigen::Matrix` arguments using EigenPy. I built a `lib` which contains `foo`. Then I import `lib` in Python and `foo` worked fine. 

However, there is a problem when I insert `import pinocchio as se3` before calling `foo` in Python: Python will crash when the execution of `foo` **terminates**, with the following error:

> Python(34796,0x7fff9fcc5340) malloc: *** error for object 0x7fc61b61d020: pointer being freed was not allocated
> *** set a breakpoint in malloc_error_break to debug
> Abort trap: 6

I also observe that the problem occurs only when the input of `foo` contains a `Eigen::Matrix` object. The weird thing to me is the execution does not stop until the function call terminates.  
Also note that the error occurs when pinocchio is imported, even if the inside of `foo` does not call any pinocchio module.

You could check out this by binding a toy function (for example the `reflex` function in issue #18 ) and import Pinocchio in Python. Many thanks for your help!

**Comment by jcarpent** [comment]
Can you provide all the code?

**Comment by zongmianli** [comment]
I have created a [minimal toy example](https://github.com/zongmianli/eigenpyIssue19
) which shows the issue.

`git clone https://github.com/zongmianli/eigenpyIssue19`
`cd eigenpyIssue19`
`mkdir build`
`cmake ..`
`make`

Then `cd ..` and `python test.py` and `python test_bad.py` and I have got these results:

> dhcp-11-61:eigenpyIssue19 zml$ python test.py 
> new function! EigenMat = 1 2 3
> dhcp-11-61:eigenpyIssue19 zml$ python test_bad.py 
> new function! EigenMat = 1 2 3
> Python(37911,0x7fff9fcc5340) malloc: *** error for object 0x7fb4a6c94540: pointer being freed was not allocated
> *** set a breakpoint in malloc_error_break to debug
> Abort trap: 6

Hope it clear and thanks in advance!

**Comment by jcarpent** [comment]
I tested your code and it works perfectly on High Sierra.
I guess the problem comes from the conflict between two installed version of EigenPy on your mac.
You can check with `locate libeigenpy.dylib`

**Comment by zongmianli** [comment]
Thanks for the help and the problem is solved. In fact I was using built-from-source Eigenpy and Homebrew Pinocchio, from which the problem comes. I did the following 
- I first `brew uninstall pinocchio` and `make uninstall` Eigenpy. Then I reinstalled and relinked both softwares via Homebrew. However this time I even got the same malloc error when running `python test.py`. 
- Then I tried to rebuild Eigenpy and Pinocchio from source and it worked.

In a word, the Homebrew version of eigenpy does not work on my machine and the from source eigenpy maybe have a problem with homebrew Pinocchio.
